File: Lv 3/하노이의 탑.py
def hanoii(answer, n, src, temp, dest):
    if n == 1:
        answer.append([src, dest])
    else:
        hanoii(answer, n-1, src, dest, temp)
        answer.append([src, dest])
        hanoii(answer, n-1, temp, src, dest)

def solution(n):
    answer = []
    hanoii(answer, n, 1, 2, 3)
    return answer

File: LV 2/하노이의 탑.java
import java.util.ArrayList;
import java.util.List;

class Solution {

    List<int[]> answer = new ArrayList<>();

    public void hanoi(int n, int src, int dest, int temp) {
        if (n > 0) {
            hanoi(n - 1, src, temp, dest);
            answer.add(new int[]{src, dest});
            hanoi(n - 1, temp, dest, src);
        }
    }

    public int[][] solution(int n) {
        hanoi(n, 1, 3, 2);
        return answer.toArray(size -> new int[0][]);
    }
}
